Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCommandEncoder.cpp

```diff
@@ -38,17 +39,14 @@
 #include "WebGPUObjectHeap.h"
 #include <WebCore/WebGPUCommandEncoder.h>
 
-#if PLATFORM(COCOA)
 #define MESSAGE_CHECK(assertion) do { \
-    if (UNLIKELY(!(assertion))) { \
-        if (auto connection = m_gpuConnectionToWebProcess.get()) \
-            connection->terminateWebProcess(); \
-        return; \
+    if (auto didPassMessageCheck = !!(assertion)) { \
+        auto connection = m_gpuConnectionToWebProcess.get(); \
+        if (!connection) \
```

Comment:
And doesn't this do nothing if the connection is null (when we would prefer to kill the WebContent process instead)?

Reply:
I guess we can't kill the WebContent process if we have no connection. Maybe we should kill the GPU process in that case, though? We're definitely Having a Bad Time (TM) if we receive an invalid message from a process we can't identify. If our goal is to fail closed, I think we do need to kill the GPU process.

Reply:
But there could be a race, right? WebContent can send an IPC to GPU process shortly before GPU process had a chance to process the last IPC.

Reply:
Indeed and bringing down the entire GPU process would be quite problematic in that case

Reply:
I guess if m_gpuConnectionToWebProcess only becomes null when the WebContent process exits, maybe that's just as good as killing the WebContent process, so there's no need to do anything?